docs/reference/scripting/grok-syntax.asciidoc

```diff
@@ -0,0 +1,231 @@
+[[grok]]
+=== Grokking grok
+Grok is a regular expression dialect that supports reusable aliased expressions. Grok works really well with syslog logs, Apache and other webserver
+logs, mysql logs, and generally any log format that is written for humans and
+not computer consumption.
+
+Grok sits on top of the https://github.com/kkos/oniguruma/blob/master/doc/RE[Oniguruma] regular expression library, so any regular expressions are
+valid in grok. Grok uses this regular expression language to allow naming
+existing patterns and combining them into more complex patterns that match your
+fields.
+
+[[grok-syntax]]
+==== Grok patterns
+The {stack} ships with numerous https://github.com/elastic/elasticsearch/blob/master/libs/grok/src/main/resources/patterns/grok-patterns[predefined grok patterns] that simplify working with grok. The syntax for reusing grok patterns
+takes one of the following forms:
+
+[%autowidth]
+|===
+|`%{SYNTAX}` | `%{SYNTAX:SEMANTIC}` |`%{SYNTAX:SEMANTIC:TYPE}`
+|===
+
+`SYNTAX`::
+The name of the pattern that will match your text. For example, `NUMBER` and
+`IP` are both patterns that are provided within the default patterns set. The
+`NUMBER` pattern matches data like `3.44`, and the `IP` pattern matches data
+like `55.3.244.1`. 
+
+`SEMANTIC`::
+The identifier you give to the piece of text being matched. For  example, `3.44`
+could be the duration of an event, so you might call it `duration`. The string
+`55.3.244.1` might identify the `client` making a request.
+
+`TYPE`::
+The data type you want to cast your named field. `int`, `long`, `double`,
+`float` and `boolean` are supported types.
```

Comment:
It's probably not worth calling out here - but this is different list than we support in runtime fields. It *does* line up with the list of things `emit` can accept though.